embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbracePendingApiCallsSender.kt

```diff
@@ -32,25 +36,38 @@ internal class EmbracePendingApiCallsSender(
         scheduledExecutorService.submit(this::scheduleApiCallsDelivery)
     }
 
-    override fun setSendMethod(sendMethod: (request: ApiRequest, payload: ByteArray) -> Unit) {
+    override fun setSendMethod(sendMethod: (request: ApiRequest, payload: ByteArray) -> ApiResponse) {
         this.sendMethod = sendMethod
     }
 
-    override fun scheduleApiCall(request: ApiRequest, payload: ByteArray) {
-        logger.logDeveloper(TAG, "Scheduling api call for retry")
-
+    override fun savePendingApiCall(request: ApiRequest, payload: ByteArray): PendingApiCall {
+        // Save the payload to disk.
         val cachedPayloadName = cacheManager.savePayload(payload)
-        val pendingApiCall = PendingApiCall(request, cachedPayloadName, clock.now())
-
-        val scheduleJob = pendingApiCalls.hasAnyPendingApiCall().not()
 
+        // Save the pending api call to disk.
+        val pendingApiCall = PendingApiCall(request, cachedPayloadName, clock.now())
         pendingApiCalls.add(pendingApiCall)
         cacheManager.savePendingApiCalls(pendingApiCalls)
+        return pendingApiCall
+    }
+
+    override fun scheduleApiCall(response: ApiResponse) {
```

Comment:
This should probably renamed now that it doesn't schedule an API call? Maybe `scheduleRetry()`?

Reply:
Yeap, totally !!